twisted/conch/client/default.py

```diff
@@ -246,18 +255,46 @@ def getPrivateKey(self):
 
     def getGenericAnswers(self, name, instruction, prompts):
         responses = []
-        try:
-            oldout, oldin = sys.stdout, sys.stdin
-            sys.stdin = sys.stdout = open('/dev/tty','r+')
+        with self._replaceStdoutStdin():
             if name:
-                print(name)
+                print(name.decode("UTF8"))
             if instruction:
-                print(instruction)
+                print(instruction.decode("UTF8"))
             for prompt, echo in prompts:
+                prompt = prompt.decode("UTF8")
                 if echo:
                     responses.append(raw_input(prompt))
                 else:
                     responses.append(getpass.getpass(prompt))
-        finally:
-            sys.stdout,sys.stdin=oldout,oldin
         return defer.succeed(responses)
+
+
+    @classmethod
+    def _openTty(cls):
+        """
+        Open /dev/tty as two streams one in read, one in write mode,
+        and return them.
```

Comment:
As per the doc standard, this needs a `@return` which describes its return value; it should be crisply documented that it returns a `TextIOWrapper` on py3 but a binary file on python 2, matching the behavior of `sys.stdin` and `sys.stdout`.  (You learn something every day, I guess!  I thought that `sys.stdin` et. al. were rather informally specified but checking up on this branch I learned that they are in fact always unicode on py3 and always bytes on py2, albeit sometimes bytes with an encoding that can accept `unicode` writes on py2.)

Speaking of accepting `unicode` writes; there's a weird nuance here that I'm not sure how to fully reproduce.  On python 2, `sys.stdin` normally has an `encoding` attribute.  Although there's a C API to set this (`PyFile_SetEncoding`) it seems like the attribute is just completely read-only from Python.  `io.open` doesn't support `encoding=` with binary mode.  It's probably fine not to care for now but this is something we should keep in mind for the future.